syntax = "proto2";

package mesop.components.uploader;

message UploadedFile {
    optional string name = 1;
    optional int32 size = 2;
    optional string mime_type = 3;
    optional bytes contents = 4;
}


message UploadEvent {
    repeated UploadedFile file = 1;
}


// Next ID: 9
message UploaderType {
    repeated string accepted_file_type = 1;
    optional string on_upload_event_handler_id = 2;
    optional string color = 3;
    optional bool disable_ripple = 4;
    optional bool disabled = 5;
    optional bool multiple = 8;

    // Type has two properties:
    // |type_index| is used for rendering
    // |type| is used for editor value
    optional int32 type_index = 6;
    optional string type = 7;
}
